### PR TITLE
create admin service

### PR DESCRIPTION
**Pull request by ekaterinaryzhenkova**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/930

### CODE REVIEW

File: Common.Configurations.targets [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  <ItemGroup Condition="($(AssemblyName.Contains('.Provider')) Or $(AssemblyName.Contains('.Models.Db'))) And $(AssemblyName.Contains('.UnitTests')) != 'true'">
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.4" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="5.0.4" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="5.0.4" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.4" />
+  </ItemGroup>
+
+  <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.49" />
```
Comment (Cheephot) [inline Common.Configurations.targets]: Update Kernel and Model to last version

File: src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AdminService.Data.Provider.MsSql.Ef
+{
+    public class AdminServiceDbContext
```
Comment (Cheephot) [inline src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs]: public class AdminServiceDbContext : DbContext, IDataProvider

**Review by Cheephot — APPROVED** [review]

File: Common.Configurations.targets (hunk 1 above)
Comment (ekaterinaryzhenkova) [reply Common.Configurations.targets]: done

File: src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs (hunk 2 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs]: done

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,25 @@
+**/.classpath
```
Comment (BelehovEgor) [inline .dockerignore]: docker ignore file in other services is different, make like those service or approve this file with Mariya or Spartak

File: src/AdminService/Program.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+{
+  public class Program
+  {
+    public static void Main(string[] args)
+    {
+      CreateHostBuilder(args).Build().Run();
+    }
+
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
```
Comment (BelehovEgor) [inline src/AdminService/Program.cs]: fix tabs

File: src/AdminService/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  "iisSettings": {
+    "windowsAuthentication": false,
+    "anonymousAuthentication": true,
+    "iisExpress": {
+      "applicationUrl": "http://localhost:29478",
+      "sslPort": 0
+    }
+  },
+  "profiles": {
+    "IIS Express": {
```
Comment (BelehovEgor) [inline src/AdminService/Properties/launchSettings.json]: do you need this block?

File: src/AdminService/AdminService.csproj [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Folder Include="Controllers\" />
```
Comment (BelehovEgor) [inline src/AdminService/AdminService.csproj]: create usual folder and remove this

File: src/AdminService/Startup.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace AdminService
+{
+  public class Startup
+  {
+    public Startup(IConfiguration configuration)
+    {
+      Configuration = configuration;
```
Comment (BelehovEgor) [inline src/AdminService/Startup.cs]: add BaseServiceInfoConfig
RabbitMQConfig, and Configure mass transit
add token middleware
add connection to db

File: src/AdminService/Startup.cs [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+      using var serviceScope = app.ApplicationServices
+        .GetRequiredService<IServiceScopeFactory>()
+        .CreateScope();
+
+      using var context = serviceScope.ServiceProvider.GetService<AdminServiceDbContext>();
+
+      context.Database.Migrate();
+     }
+
+     private string HidePassord(string line)
```
Comment (Cheephot) [inline src/AdminService/Startup.cs]: use Kernel

File: src/AdminService/Properties/launchSettings.json (hunk 5 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService/Properties/launchSettings.json]: fixed

File: src/AdminService/AdminService.csproj (hunk 6 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService/AdminService.csproj]: fixed

File: src/AdminService/Startup.cs (hunk 7 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService/Startup.cs]: done

File: .dockerignore (hunk 3 above)
Comment (ekaterinaryzhenkova) [reply .dockerignore]: done

File: src/AdminService/Program.cs (hunk 4 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService/Program.cs]: done

File: src/AdminService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Configurations;
+using LT.DigitalOffice.Models.Broker.Requests.User;
+
+namespace AdminService.Models.Dto.Configuration
+{
+  public class RabbitMqConfig : BaseRabbitMqConfig
+  {
+    public string GetSmtpCredentialsEndpoint { get; set; }
```
Comment (MariyaSin) [inline src/AdminService.Models.Dto/Configuration/RabbitMqConfig.cs]: replase to UpdateSmtpCredentialsEndpoint

File: src/AdminService/Program.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Serilog;
```
Comment (MariyaSin) [inline src/AdminService/Program.cs]: use editor config reset tabs

File: src/AdminService/Program.cs [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace AdminService
```
Comment (MariyaSin) [inline src/AdminService/Program.cs]: LT.DigitalOffice.AdminService

File: src/AdminService/Startup.cs [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Serilog;
+using StackExchange.Redis;
+
+namespace AdminService
```
Comment (MariyaSin) [inline src/AdminService/Startup.cs]: LT.DigitalOffice.AdminService

File: src/AdminService/appsettings.json [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  "RabbitMQ": {
+
+    // CRService
+    "CheckUserRightsEndpoint": "CRService_CheckUserRights",
+
+    // AuthService
+    "ValidateTokenEndpoint": "AuthService_ValidateToken",
+
+    //AdminService
+    "CreateAdminEndpoint": "AdminService_CreateAdmin",
```
Comment (MariyaSin) [inline src/AdminService/appsettings.json]: "CreateAdminEndpoint": "UserService_CreateAdmin"

File: src/AdminService/appsettings.json [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  "RabbitMQ": {
+
+    // CRService
+    "CheckUserRightsEndpoint": "CRService_CheckUserRights",
+
+    // AuthService
+    "ValidateTokenEndpoint": "AuthService_ValidateToken",
+
+    //AdminService
```
Comment (MariyaSin) [inline src/AdminService/appsettings.json:40]: //UserService
User service creates and saves admin data.

File: src/AdminService/appsettings.json [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    "CreateAdminEndpoint": "AdminService_CreateAdmin",
+
+    "GetSmtpCredentialsEndpoint": "AdminService_GetSmtpCredentials",
```
Comment (MariyaSin) [inline src/AdminService/appsettings.json:41]: // EmailService
    "UpdateSmtpCredentialsEndpoint": "EmailService_UpdateSmtpCredentials"

File: src/AdminService/appsettings.json [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+    // UserService
+    "CheckUserIsAdminEndpoint": "UserService_GetUserAdminStatus"
```
Comment (MariyaSin) [inline src/AdminService/appsettings.json]: insert to 42 line

File: .github/workflows/main.yml [hunk 17]
```diff
@@ -0,0 +1,36 @@
+name: Build & Test ProjectService
```
Comment (asyagraf) [inline .github/workflows/main.yml]: mb AdminService?

File: src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs [hunk 18]
```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace AdminService.Data.Provider.MsSql.Ef
+{
+    public class AdminServiceDbContext : DbContext, IDataProvider
```
Comment (asyagraf) [inline src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs]: fix tabs (lines 5-11)

File: src/AdminService.Data.Provider/IDataProvider.cs [hunk 19]
```diff
@@ -0,0 +1,10 @@
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Enums;
+
+namespace AdminService.Data.Provider
+{
+  [AutoInject(InjectType.Scoped)]
+  public interface IDataProvider
```
Comment (asyagraf) [inline src/AdminService.Data.Provider/IDataProvider.cs]: maybe IDataProvider should inherit IBaseDataProvider

File: .github/workflows/main.yml (hunk 17 above)
Comment (ekaterinaryzhenkova) [reply .github/workflows/main.yml]: done

File: src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs (hunk 18 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService.Data.Provider.MsSql.Ef/AdminServiceDbContext.cs]: done

File: src/AdminService.Data.Provider/IDataProvider.cs (hunk 19 above)
Comment (ekaterinaryzhenkova) [reply src/AdminService.Data.Provider/IDataProvider.cs]: add

**Review by MariyaSin — APPROVED** [review]

**Review by grinakate — APPROVED** [review]

**Review by asyagraf — APPROVED** [review]

**Review by irlirla — APPROVED** [review]

**Pull request merged**